packages/jira-adapter/src/change_validators/workflowsV2/outbound_transition.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2024 Salto Labs Ltd.
+ * Licensed under the Salto Terms of Use (the "License");
+ * You may not use this file except in compliance with the License.  You may obtain a copy of the License at https://www.salto.io/terms-of-use
+ *
+ * CERTAIN THIRD PARTY SOFTWARE MAY BE CONTAINED IN PORTIONS OF THE SOFTWARE. See NOTICE FILE AT https://github.com/salto-io/salto/blob/main/NOTICES
+ */
+import { collections, values } from '@salto-io/lowerdash'
+import {
+  ChangeValidator,
+  SeverityLevel,
+  getChangeData,
+  isInstanceChange,
+  isAdditionOrModificationChange,
+  ReferenceExpression,
+  isReferenceExpression,
+} from '@salto-io/adapter-api'
+import { isWorkflowV2Instance, WorkflowTransitionLinks, WorkflowV2Instance } from '../../filters/workflowV2/types'
+
+let fromStatusReference: string | ReferenceExpression | undefined
+let transitionName: string
+let transitionType: string
+let fromStatusReferenceName: string | undefined
+const { isDefined } = values
+const { awu } = collections.asynciterable
+
+const addTransitionToMap = (
+  map: Map<string | ReferenceExpression, Set<string>>,
+  key: string | ReferenceExpression,
+  name: string,
+): boolean => {
+  if (map.has(key) && map.get(key)?.has(name)) {
+    return false
+  }
+  if (!map.has(key)) {
+    map.set(key, new Set<string>())
+  }
+  map.get(key)?.add(name)
+  return true
+}
```

Comment:
I think this will be cleaner. There is no reason to add an empty set to the map and then get it to add an element to it.
```suggestion
const addTransitionToMap = (
  map: Map<string | ReferenceExpression, Set<string>>,
  key: string | ReferenceExpression,
  name: string,
): boolean => {
  const keySet = map.get(key)
  if (keySet !== undefined) {
    return keySet.has(name)
  }
  const addedSet = new Set<string>([name])
  map.set(key, addedSet)
  return true
}
```

Reply:
Thanks i will try to make it clearer, 
your implementation does not cover all cases (key can have more than one name)